hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/sigs/HollowScreening.java

```diff
@@ -0,0 +1,183 @@
+import com.hedera.node.app.service.mono.sigs.utils.MiscCryptoUtils;
+import com.hedera.node.app.service.mono.utils.EntityNum;
+import com.hedera.node.app.service.mono.utils.PendingCompletion;
+import com.hedera.node.app.service.mono.utils.RationalizedSigMeta;
+import com.hedera.node.app.service.mono.utils.accessors.SwirldsTxnAccessor;
+import com.swirlds.common.crypto.SignatureType;
+import com.swirlds.common.crypto.TransactionSignature;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.tuweni.bytes.Bytes;
+
+/**
+ * A utility class for {@link Expansion} and {@link Rationalization} that, given a list of {@link TransactionSignature}
+ * and infrastructure for looking up accounts, is able to construct a list of {@link PendingCompletion} which may be
+ * performed in the given {@link SwirldsTxnAccessor}, if all key/sig activation validations pass.
+ *
+ * <p>Also exposes methods that, given {@link JWildcardECDSAKey}s as input, replace the {@link JWildcardECDSAKey}/s with their
+ * corresponding {@link JECDSASecp256k1Key}, if such was present in the list of parsed {@link TransactionSignature}s.
+ * This process is also referred to as "de-hollowing". The {@code payerReqSig} and {@code otherReqSigs} contained
+ * in the {@link RationalizedSigMeta} must be free of
+ * {@link JWildcardECDSAKey}s, so that subsequent key activation tests pass, so this de-hollow process is essential.
+ */
+public class HollowScreening {
+
+    static boolean atLeastOneWildcardECDSAKeyIn(@NonNull final JKey payerKey, @NonNull List<JKey> otherPartyKeys) {
+        if (payerKey.hasWildcardECDSAKey()) {
+            return true;
+        }
+        for (final var otherReqKey : otherPartyKeys) {
+            if (otherReqKey.hasWildcardECDSAKey()) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Encapsulates the result of a hollow screening.
+     *
+     * @param pendingCompletions a list of {@link PendingCompletion} if there the hollow screening found possible
+     *                           hollow account completions. {@code null} otherwise
+     * @param replacedPayerKey a {@link JECDSASecp256k1Key} if payer key was {@link JWildcardECDSAKey} and a corresponding
+     * {@link JECDSASecp256k1Key} was found in the txn sigs. {@code null} otherwise
+     * @param replacedOtherKeys a list of all other req keys, where possible {@link JWildcardECDSAKey} -> {@link JECDSASecp256k1Key}
+     *                          conversions were made; {@code null} if no conversions were made
+     */
+    record HollowScreenResult(
+            @Nullable List<PendingCompletion> pendingCompletions,
+            @Nullable JKey replacedPayerKey,
+            @Nullable List<JKey> replacedOtherKeys) {}
```

Comment:
Going back to first principles --- do we really need to replace any `JWildcardECDSAKey` with its corresponding `JEcdsaSecp256k1Key` in the req keys, as we've been doing?

AFAICT, we can leave the `JWildcardECDSAKey` in the `RationalizedSigMeta` required keys. [At the point of constructing the pkToSig function in  ](https://github.com/hashgraph/hedera-services/blob/bfd8d9bbb9720fd6da002a2f4c1d55593c75cf1e/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/keys/HederaKeyActivation.java#L133-L141) `HederaKeyActivation`, we can add something along the lines of:
```
if (pk.length == EVM_ADDRESS_SIZE && sig.getSignatureType() == SignatureType.ECDSA_SECP256K1) {
    final var evmAddress = MiscCryptoUtils.extractEvmAddressFromDecompressedECDSAKey(sig.getExpandedPublicKeyDirect());
    if (Arrays.equals(pk, evmAddress)) {
        return sig;
    }
}
```
[and in isActive() ](https://github.com/hashgraph/hedera-services/blob/bfd8d9bbb9720fd6da002a2f4c1d55593c75cf1e/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/keys/HederaKeyActivation.java#L104-L124)instead of immediately returning  `INVALID_MISSING_SIG`  when we see a `JWildcardECDSAKey`, we can add:
```
else if (key.hasHollowKey()) {
    return validity.test(key, sigsFn.apply(key.getHollowKey().getEvmAddress()));
}
```

thus removing the need for replacing `JWildcardECDSAKey` and removing/clearing up a lot of code (e.g. [here, in prechecks)](https://github.com/hashgraph/hedera-services/blob/bfd8d9bbb9720fd6da002a2f4c1d55593c75cf1e/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/sigs/verification/PrecheckVerifier.java#L71-L84)


Of course, we are going to be executing a little more logic in `handle`(), but we are already going through all of the sigs anyway in `pkToSigsMapFrom`, and we will be comparing arrays only for `JWildcardECDSAKey` 🤔 

wdyt @tinker-michaelj  @Neeharika-Sompalli

Reply:
That's a great observation @dimitar-dinev; we should definitely keep it in mind for the re-factored version of the Signature & Keys facility that will live in `hedera-app`. 💯